features/graph/api/src/main/java/org/opennms/netmgt/graph/api/generic/GenericVertex.java

```diff
@@ -50,6 +50,7 @@ public GenericVertex(GenericVertex copyMe){
     }
 
 
+    // TODO MVR this is a duplicat eand should probably be removed, as a vertex is already a vertexref, there is no need to have a `getVertexRef` method
     public GenericVertexRef getVertexRef() {
```

Comment:
@mvrueden we should discuss this.
I change that: The GenericVertexRef is a immutable final object - it is a composite key at the end and not the mother class of GenericVertex.
The advantage is that we can pass around that key without having the problem that some one casts it to a GenericVertex or changes things. I would argue that this makes a better system design - but let's discuss

Reply:
I can see your point. If we want to do it that way, we should remove the inheritance and name the method `asVertexRef`. If we keep the inheritance, the vertex ref should be created using a copy constructor. However both is out of the scope of this issue/PR.